enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,489 @@
+---
+title: prevent-user-workloads-from-being-scheduled-on-control-plane-nodes
+authors:
+  - knelasevero
+  - ingvagabund
+  - flavianmissi
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - ingvagabund
+  - deads2k
+  - flavianmissi
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - deads2k
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - deads2k
+creation-date: 2024-04-01
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPSTRAT-790
+  - https://issues.redhat.com/browse/WRKLDS-1015
+  - https://issues.redhat.com/browse/WRKLDS-1060
+see-also:
+  - .
+---
+
+To get started with this template:
+
+1. **Pick a domain.** Find the appropriate domain to discuss your enhancement.
+1. **Make a copy of this template.** Copy this template into the directory for
+   the domain.
+1. **Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+1. **Merge after reaching consensus.** Merge when there is consensus
+   that the design is complete and all reviewer questions have been
+   answered so that work can begin. Come back and update the document
+   if important details (API field names, workflow, etc.) change
+   during code review.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Prevent user workloads from being scheduled on control plane nodes
+
+## Summary
```

Comment:
I didnt read the who enhancement, just verifying that :
1. compact cluster will work with schedulable masters
2. compact cluster will be extendable with more workers and master can be schedulable or unscheduble post extention
3. 4 nodes cluster will work with schedulable masters
4.

Reply:
This won't be enforced out-of-the-box. Customer needs to create a VAPBinding and do the other manual steps.

In any way, anything that needs to be scheduled on master nodes need to tolerate the `node-role.kubernetes.io/control-plane:NoExecute` taint and new rules (with custom verbs and resource keys) need to be added so if VAP is enforced pods tolerating these taints can be applied:

```
rules:
- apiGroups: [""]
  resources: ["node-role.kubernetes.io/control-plane"]
  verbs: ["NoExecute"]
```

Then, yes, if VAP is enforced, all should be fine.